### ISSUE TITLE
Server breaks KeePass auto-type feature

### ISSUE THREAD

**Issue by RePlay-be** [symless/synergy-core#4400]
Synergy was configured as server, but stopping the server did not help either. I'm running Windows 7 x64. I'm using a Lenovo Y580 laptop.

[KeePass](http://keepass.info/) is a free password manager. One of the features is that it can fill out login forms for you. I don't know the technical details, but I believe it simulates keystrokes. Essentially it types it out for you. More information about the feature can be found [here](http://keepass.info/help/base/autotype.html).

Without the Synergy service it types the following: {USERNAME}{TAB}{PASSWORD}{ENTER}. With the service enabled it looks like it's doing the same thing but no output is produced. If you change the focus whilst it's typing to something that doesn't take input, you can hear the Windows error dings for the remaining characters it's typing.

To reproduce this:
1. Install KeePass
2. Make sure the Synergy service is enabled
3. Add a password entry to KeePass and make sure to use a long enough password for easy testing (say 60 characters).
4. Right click the saved password entry and select auto-type (or press CTRL+V).
5. KeePass should drop into the background and attempt to perform the auto-type.

**Comment by beporter** [comment]
The last issue is likely a duplicate of (#3823 / #4057).

Try using the steps in this comment (https://github.com/synergy/synergy/issues/4057#issuecomment-67515368) to see if you can reproduce the stuck keys that way. (I don't think that bug is Mac-specific.)

**Comment by RePlay-be** [comment]
I've tried the steps from that comment but couldn't reproduce the problem. It's worth mentioning that I'm not connected to any clients.

I have another piece of information however. The problems with the volume control and the KeePass auto-type feature only occur when **Elevate mode is enabled**.

**Comment by nbolton** [comment]
@RePlay-be Would you mind moving these issues to another thread? Thanks.

### Control of the volume with the functions keys on my laptop reverts to Windows

The [Energy Management software](https://support.lenovo.com/ca/en/products/laptops-and-netbooks/ideapad-y-series-laptops/ideapad-y580/downloads/DS029705) for my Lenovo laptop also includes control of the function keys of my laptop, like controlling the brightness and volume, disabling the touchpad, etc. An OSD is shown when pressing these buttons. Changing the volume with this tool installed changes the volume in steps of 10%.

With the Synergy service enabled, control of the volume is reverted back to Windows. Volume changes in steps of 1-2% and the OSD is not shown anymore. Other functions like brightness control still work as before.


### Shift or control are seemingly randomly enabled

In effect, it was like CTRL or ALT was being held. The Windows on-screen keyboard did not show the keys being pressed I believe. Pressing CTRL or ALT manually fixed the problem.
I could not find a way to reproduce this, and it did not occur very often either: maximum once or twice in a day.

**Comment by stevechk** [comment]
I have the same issue with Synergy server running on Win 7 64-bit. When Synergy is active, the auto-type feature in KeePass doesn't work. After opening the control panel and hitting "stop" to stop the service, the feature starts working again. 

FWIW, Keepass appears to use SendInput (https://msdn.microsoft.com/en-us/library/windows/desktop/ms646310%28v=vs.85%29.aspx) to send the keystrokes on the Windows platform.

**Comment by gregorko** [comment]
I can still see this problem with Synergy 1.7.6, when the Synergy service is running, without being connected, KeePass auto-type feature fails to work.

**Comment by xpac1985** [comment]
Same issue here with Synergy 1.7.6. and KeePass 2.32.
KeePass is not run in elevated mode, but the secure desktop is used to enter the main database password (if that makes any differences).
I also got issues with Synergy running while playing Stardew Valley - the mouse works fine, until the first time a key is pressed. Any keys besides ESC stop working after that. Doing Alt+Tab out of the game and back in solves the problem, until the next time a key is pressed.
Only stopping the synergy service before starting the game solves the problem.
Funnily, starting the Synergy service while already running the game doesn't cause any problems.

I know, the game bug is a different topic, but maybe they have same basic issue, so maybe that helps.

**Comment by lawre** [comment]
i also have the keepass autotype not working problem.  If I set the Synergy service to Automatic (Delayed Start), the problem doesn't seem to present itself.

**Comment by xpac1985** [comment]
The **Delayed Start** thing helped me too, but only in 80% of cases.
Sometimes I still need to stop & start the service, then everything works...

**Comment by lawre** [comment]
xpac1985: you are correct, I have seen the problem re-present itself even with the Delayed Start.

**Comment by ccoenen** [comment]
@beporter's link is no longer working as the organisation has been renamed in the meantime. His  Comment can be found here: https://github.com/symless/synergy/issues/4057#issuecomment-67515368

**Comment by ccoenen** [comment]
I have this same problem. As long as Synergy Server is running, AutoType of KeePass does not work. If I go to window's services panel (type `services.msc` into any start menu since windows vista) and restart the synergy service, it starts working alright.

**Comment by ccoenen** [comment]
Nope. Sorry to report that tarting the service delayed does not fix the problem. This is becoming quite annoying.

**Comment by ccoenen** [comment]
This may be another hint: KeePass just failed me *mid-password*. So It used to work. Then I Auto-Typed, it worked for three characters, then it stopped auto-typing.

After restarting the synergy service, it was back to normal (working auto-type).

Maybe it's the typing speed that throws synergy off?

This was with no clients connected on a Win10 with Synergy configured as a server.

**Comment by lalminana** [comment]
I just bought it and hit this year old bug, keepass doesn't work and stardew valley fails the same way it does for xpac1985, this sucks.

**Comment by posterberg** [comment]
Yup, this is the main reason why I stopped using Synergy...

**Comment by xpac1985** [comment]
On Windows 10 Enterprise with current Synergy (1.8.8), it seems like I don't have that problem anymore...

**Comment by posterberg** [comment]
I need to test again in that case, good news =)

**Comment by ccoenen** [comment]
I also can't remember when I noticed this the last time. _seems_ to be gone with 1.8.7 or 1.8.8 (perhaps even earlier?)

**Comment by lawre** [comment]
I would also like to report this issue seems to be gone after upgrading to Windows 10
